src/test/java/com/wikia/webdriver/elements/mercury/components/Search.java

```diff
@@ -86,4 +78,79 @@ public Search clickClearSearchButton() {
     return this;
   }
 
+  public SearchResultsPage clickEnterAndNavigateToSearchResults() {
+    new Actions(driver).sendKeys(Keys.ENTER).perform();
+
+    return new SearchResultsPage();
+  }
+
+  public boolean isInputFieldSearchIconVisible() {
+    try {
+      wait.forElementVisible(inputFieldSearchIcon);
+      return inputFieldSearchIcon.isDisplayed();
+    } catch (NoSuchElementException e) {
+      PageObjectLogging.logInfo(e.getMessage());
+      return false;
+    }
+  }
+
+  public boolean isClearSearchButtonVisible() {
+    try {
+      wait.forElementVisible(clearSearchButton);
+      return clearSearchButton.isDisplayed();
+    } catch (NoSuchElementException e) {
+      PageObjectLogging.logInfo(e.getMessage());
+      return false;
+    }
+  }
+
+  public boolean isSearchInputFieldVisible() {
+    try {
+      wait.forElementVisible(searchInput);
+      return searchInput.isDisplayed();
+    } catch (NoSuchElementException e) {
+      PageObjectLogging.logInfo(e.getMessage());
+      return false;
+    }
+  }
+
+  public boolean isSearchInputFieldEditable() {
+    try {
+      wait.forElementClickable(searchInput, 0).click();
+      return true;
+    } catch (NoSuchElementException e) {
+      PageObjectLogging.logInfo(e.getMessage());
+      return false;
+    } catch (WebDriverException e) {
+      // @TODO: this would be best pushed to Selenium upstream to actually throw a more specific
+      // exception
+      if (e.getMessage().contains("Element is not clickable at point")) {
+        return false;
+      } else {
+        throw e;
+      }
+    }
+  }
+
+  public boolean areSearchSuggestionsDisplayed() {
+    try {
+      wait.forElementClickable(By.cssSelector(searchSuggestionClass),
+                               SUGGESTIONS_TIMEOUT_IN_SECONDS);
+      return true;
+    } catch (TimeoutException e) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-s1/coding_rules#rule_key=squid%3AS1166)